### ISSUE TITLE
Installing `egon.data` pulls in incompatible `marshmallow` versions

### ISSUE THREAD

**Issue by wheitkoetter** [openego/eGon-data#36]
@gplssm : there is already @gnn working on this issue

When installing eGon-data, Amelia and I get the following error message:

marshmallow-sqlalchemy 0.24.0 requires marshmallow>=3.0.0, but you'll have marshmallow 2.21.0 which is incompatible.

Did anyone else experience this error or do you have a recommendation how to solve it?

**Comment by AmeliaNadal** [comment]
The wohle output message when installing eGon-data:
> pip install -e eGon-data
Obtaining file:///home/nada_am/eGon-data
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
ERROR: Exception:
Traceback (most recent call last):
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/cli/base_command.py", line 228, in _main
    status = self.run(options, args)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/cli/req_command.py", line 182, in wrapper
    return func(self, options, args)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/commands/install.py", line 324, in run
    reqs, check_supported_wheels=not options.target_dir
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/resolution/legacy/resolver.py", line 183, in resolve
    discovered_reqs.extend(self._resolve_one(requirement_set, req))
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/resolution/legacy/resolver.py", line 388, in _resolve_one
    abstract_dist = self._get_abstract_dist_for(req_to_install)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/resolution/legacy/resolver.py", line 326, in _get_abstract_dist_for
    return self.preparer.prepare_editable_requirement(req)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/operations/prepare.py", line 531, in prepare_editable_requirement
    req.check_if_exists(self.use_user_site)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/req/req_install.py", line 433, in check_if_exists
    existing_dist = get_distribution(self.req.name)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_internal/utils/misc.py", line 531, in get_distribution
    pkg_resources.working_set.require(req_name)
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages/pip/_vendor/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pip._vendor.pkg_resources.ContextualVersionConflict: (marshmallow 2.21.0 (/home/nada_am/anaconda3/envs/egondata_env/lib/python3.6/site-packages), Requirement.parse('marshmallow>=3.0.0'), {'marshmallow-sqlalchemy'})

**Comment by AmeliaNadal** [comment]
(egondata_env) nada_am@torch:~$ pip install -e eGon-data
Obtaining file:///home/nada_am/eGon-data
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: apache-airflow in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from egon.data==0.0.0) (1.10.12)
Requirement already satisfied: click in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from egon.data==0.0.0) (7.1.2)
Requirement already satisfied: oedialect==0.0.8 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from egon.data==0.0.0) (0.0.8)
Requirement already satisfied: sqlalchemy~=1.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.3.20)
Requirement already satisfied: configparser<3.6.0,>=3.5.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (3.5.3)
Requirement already satisfied: cattrs~=1.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.0.0)
Requirement already satisfied: python-dateutil<3,>=2.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.8.1)
Requirement already satisfied: flask-caching<1.4.0,>=1.3.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.3.3)
Requirement already satisfied: markdown<3.0,>=2.5.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.6.11)
Requirement already satisfied: funcsigs<2.0.0,>=1.0.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.0.2)
Requirement already satisfied: jinja2<2.12.0,>=2.10.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.11.2)
Requirement already satisfied: jsonschema~=3.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (3.2.0)
Requirement already satisfied: python-daemon>=2.1.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.2.4)
Requirement already satisfied: colorlog==4.0.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (4.0.2)
Requirement already satisfied: psutil<6.0.0,>=4.2.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (5.7.3)
Requirement already satisfied: flask-login<0.5,>=0.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.4.1)
Requirement already satisfied: requests<3,>=2.20.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.24.0)
Requirement already satisfied: graphviz>=0.12 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.14.2)
Requirement already satisfied: future<0.19,>=0.16.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.18.2)
Requirement already satisfied: werkzeug<1.0.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.16.1)
Requirement already satisfied: email-validator in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.1.1)
Requirement already satisfied: python-slugify<5.0,>=3.0.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (4.0.1)
Requirement already satisfied: dill<0.4,>=0.2.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.3.3)
Requirement already satisfied: alembic<2.0,>=1.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.4.3)
Requirement already satisfied: flask-admin==1.5.4 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.5.4)
Requirement already satisfied: unicodecsv>=0.14.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.14.1)
Requirement already satisfied: flask-wtf<0.15,>=0.14.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.14.3)
Requirement already satisfied: attrs~=19.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (19.3.0)
Requirement already satisfied: gunicorn<21.0,>=19.5.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (20.0.4)
Requirement already satisfied: flask-appbuilder~=2.2; python_version >= "3.6" in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.3.4)
Requirement already satisfied: cached-property~=1.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.5.2)
Requirement already satisfied: setproctitle<2,>=1.1.8 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.1.10)
Requirement already satisfied: tzlocal<2.0.0,>=1.4 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.5.1)
Requirement already satisfied: argcomplete~=1.10 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.12.1)
Requirement already satisfied: tenacity==4.12.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (4.12.0)
Requirement already satisfied: sqlalchemy-jsonfield~=0.9; python_version >= "3.5" in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.9.0)
Requirement already satisfied: pandas<2.0,>=0.17.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.1.4)
Requirement already satisfied: python-nvd3~=0.15.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.15.0)
Requirement already satisfied: iso8601>=0.1.12 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.1.13)
Requirement already satisfied: flask<2.0,>=1.1.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.1.2)
Requirement already satisfied: tabulate<0.9,>=0.7.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.8.7)
Requirement already satisfied: croniter<0.4,>=0.3.17 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.3.35)
Requirement already satisfied: pendulum==1.4.4 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.4.4)
Requirement already satisfied: flask-swagger<0.3,>=0.2.13 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.2.14)
Requirement already satisfied: json-merge-patch==0.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.2)
Requirement already satisfied: zope.deprecation<5.0,>=4.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (4.4.0)
Requirement already satisfied: typing-extensions>=3.7.4; python_version < "3.8" in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (3.7.4.3)
Requirement already satisfied: lazy-object-proxy~=1.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (1.5.1)
Requirement already satisfied: pygments<3.0,>=2.0.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (2.7.2)
Requirement already satisfied: thrift>=0.9.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from apache-airflow->egon.data==0.0.0) (0.13.0)
Requirement already satisfied: psycopg2-binary in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from oedialect==0.0.8->egon.data==0.0.0) (2.8.6)
Requirement already satisfied: geoalchemy2<0.7.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from oedialect==0.0.8->egon.data==0.0.0) (0.6.3)
Requirement already satisfied: shapely in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from oedialect==0.0.8->egon.data==0.0.0) (1.7.1)
Requirement already satisfied: six>=1.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from python-dateutil<3,>=2.3->apache-airflow->egon.data==0.0.0) (1.15.0)
Requirement already satisfied: MarkupSafe>=0.23 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from jinja2<2.12.0,>=2.10.1->apache-airflow->egon.data==0.0.0) (1.1.1)
Requirement already satisfied: importlib-metadata; python_version < "3.8" in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from jsonschema~=3.0->apache-airflow->egon.data==0.0.0) (2.0.0)
Requirement already satisfied: setuptools in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from jsonschema~=3.0->apache-airflow->egon.data==0.0.0) (49.6.0.post20201009)
Requirement already satisfied: pyrsistent>=0.14.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from jsonschema~=3.0->apache-airflow->egon.data==0.0.0) (0.17.3)
Requirement already satisfied: docutils in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from python-daemon>=2.1.1->apache-airflow->egon.data==0.0.0) (0.16)
Requirement already satisfied: lockfile>=0.10 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from python-daemon>=2.1.1->apache-airflow->egon.data==0.0.0) (0.12.2)
Requirement already satisfied: chardet<4,>=3.0.2 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from requests<3,>=2.20.0->apache-airflow->egon.data==0.0.0) (3.0.4)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from requests<3,>=2.20.0->apache-airflow->egon.data==0.0.0) (1.25.11)
Requirement already satisfied: certifi>=2017.4.17 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from requests<3,>=2.20.0->apache-airflow->egon.data==0.0.0) (2020.6.20)
Requirement already satisfied: idna<3,>=2.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from requests<3,>=2.20.0->apache-airflow->egon.data==0.0.0) (2.10)
Requirement already satisfied: dnspython>=1.15.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from email-validator->apache-airflow->egon.data==0.0.0) (2.0.0)
Requirement already satisfied: text-unidecode>=1.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from python-slugify<5.0,>=3.0.0->apache-airflow->egon.data==0.0.0) (1.3)
Requirement already satisfied: Mako in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from alembic<2.0,>=1.0->apache-airflow->egon.data==0.0.0) (1.1.3)
Requirement already satisfied: python-editor>=0.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from alembic<2.0,>=1.0->apache-airflow->egon.data==0.0.0) (1.0.4)
Requirement already satisfied: wtforms in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-admin==1.5.4->apache-airflow->egon.data==0.0.0) (2.3.3)
Requirement already satisfied: itsdangerous in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-wtf<0.15,>=0.14.2->apache-airflow->egon.data==0.0.0) (1.1.0)
Requirement already satisfied: Flask-OpenID<2,>=1.2.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (1.2.5)
Requirement already satisfied: apispec[yaml]<2,>=1.1.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (1.3.3)
Requirement already satisfied: Flask-Babel<2,>=1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (1.0.0)
Requirement already satisfied: Flask-SQLAlchemy<3,>=2.4 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (2.4.4)
Requirement already satisfied: prison<1.0.0,>=0.1.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (0.1.3)
Requirement already satisfied: PyJWT>=1.7.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (1.7.1)
Requirement already satisfied: colorama<1,>=0.3.9 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (0.4.4)
Requirement already satisfied: sqlalchemy-utils<1,>=0.32.21 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (0.36.8)
Requirement already satisfied: Flask-JWT-Extended<4,>=3.18 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (3.24.1)
Collecting marshmallow<3.0.0,>=2.18.0
  Using cached marshmallow-2.21.0-py2.py3-none-any.whl (50 kB)
Requirement already satisfied: marshmallow-sqlalchemy<1,>=0.16.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (0.24.0)
Requirement already satisfied: marshmallow-enum<2,>=1.4.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (1.5.1)
Requirement already satisfied: pytz in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from tzlocal<2.0.0,>=1.4->apache-airflow->egon.data==0.0.0) (2020.4)
Requirement already satisfied: typing>=3.6; python_version < "3.7" in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from sqlalchemy-jsonfield~=0.9; python_version >= "3.5"->apache-airflow->egon.data==0.0.0) (3.7.4.3)
Requirement already satisfied: numpy>=1.15.4 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from pandas<2.0,>=0.17.1->apache-airflow->egon.data==0.0.0) (1.19.3)
Requirement already satisfied: natsort in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from croniter<0.4,>=0.3.17->apache-airflow->egon.data==0.0.0) (7.0.1)
Requirement already satisfied: pytzdata>=2018.3.0.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from pendulum==1.4.4->apache-airflow->egon.data==0.0.0) (2020.1)
Requirement already satisfied: PyYAML>=5.1 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from flask-swagger<0.3,>=0.2.13->apache-airflow->egon.data==0.0.0) (5.3.1)
Requirement already satisfied: zipp>=0.5 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from importlib-metadata; python_version < "3.8"->jsonschema~=3.0->apache-airflow->egon.data==0.0.0) (3.4.0)
Requirement already satisfied: python3-openid>=2.0 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from Flask-OpenID<2,>=1.2.5->flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (3.2.0)
Requirement already satisfied: Babel>=2.3 in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from Flask-Babel<2,>=1->flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (2.8.0)
Requirement already satisfied: defusedxml in ./anaconda3/envs/egondata_env/lib/python3.6/site-packages (from python3-openid>=2.0->Flask-OpenID<2,>=1.2.5->flask-appbuilder~=2.2; python_version >= "3.6"->apache-airflow->egon.data==0.0.0) (0.6.0)
Installing collected packages: egon.data, marshmallow
  Attempting uninstall: egon.data
    Found existing installation: egon.data 0.0.0
    Uninstalling egon.data-0.0.0:
      Successfully uninstalled egon.data-0.0.0
  Running setup.py develop for egon.data
  Attempting uninstall: marshmallow
    Found existing installation: marshmallow 3.9.0
    Uninstalling marshmallow-3.9.0:
      Successfully uninstalled marshmallow-3.9.0
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

marshmallow-sqlalchemy 0.24.0 requires marshmallow>=3.0.0, but you'll have marshmallow 2.21.0 which is incompatible.
Successfully installed egon.data marshmallow-2.21.0

**Comment by gnn** [comment]
Tldr.: Use `pip install --use-feature=2020-resolver`.

Long Version:
OK. So I did some dependency chasing and it turns out the reason for this error is `pip`'s insufficient dependency resolution. Apache Airflow depends on `flask-appbuilder` which depends on `marshmallow` in the version range `[2.18.0, 3.0.0)`. But it also depends on `marshmallow-sqlalchemy` in the version range `[0.16.1, 1)`. The `marshmallow-sqlalchemy` package also depends on `marshmallow` and it released version `0.24.0`, which bumped the lower version constraint on `marshmallow` to `>= 3.0.0`, around two weeks ago. When looking for dependencies, `pip` doesn't fully solve the dependency constraints. It simply takes the newest eligible `marshmallow-sqlalchemy`, which, being version `0.24.0`, creates the version constraint `marshmallow >= 3.0.0` which is incompatible with `flask-appbuilder`'s own `marshmallow < 3.0.0`. The fact that version `0.23.1` of `marshmallow-sqlalchemy` is eligible and doesn't create the constraint incompatibility isn't something that `pip` is aware of with its current dependency resolution algorithm. Fortunately they fixed that and you can get the new algorithm by specifying `--use-feature=2020-resolver` when installing `egon.data` which finds a correct solution and doesn't lead to the version conflict.

I'll fix this issue, once there's a note about the `pip` flag in the installation documentation.

**Comment by AmeliaNadal** [comment]
Thanks a lot, the installation ran without an error message!
